tests/components/dynalite/test_config_flow.py

```diff
@@ -1,36 +1,91 @@
-                )
-                mock_create.assert_called_once()
-                assert mock_create.mock_calls[0] == call(
-                    title=host, data={"host": host}
-                )
+from asynctest import patch
+
+from homeassistant import config_entries
+from homeassistant.components import dynalite
+
+
+async def run_flow(hass, setup, connection):
+    """Run a flow with or without errors and return result."""
+    host = "1.2.3.4"
+    with patch(
+        "homeassistant.components.dynalite.bridge.DynaliteDevices.async_setup",
+        return_value=setup,
+    ), patch(
+        "homeassistant.components.dynalite.bridge.DynaliteDevices.available", connection
+    ), patch(
+        "homeassistant.components.dynalite.bridge.CONNECT_INTERVAL", 0
+    ):
+        result = await hass.config_entries.flow.async_init(
+            dynalite.DOMAIN,
+            context={"source": config_entries.SOURCE_IMPORT},
+            data={dynalite.CONF_HOST: host},
+        )
+    return result
+
+
+async def test_flow_works(hass):
+    """Test a successful config flow."""
+    result = await run_flow(hass, True, True)
+    assert result["type"] == "create_entry"
+
+
+async def test_flow_setup_fails(hass):
+    """Test a flow where async_setup fails."""
+    result = await run_flow(hass, False, True)
+    assert result["type"] == "abort"
+    assert result["reason"] == "bridge_setup_failed"
+
+
+async def test_flow_no_connection(hass):
+    """Test a flow where connection times out."""
+    result = await run_flow(hass, True, False)
+    assert result["type"] == "abort"
+    assert result["reason"] == "no_connection"
+
+
+async def test_existing_same_config(hass):
+    """Test when the entry exists with the same config."""
+    host = "1.2.3.4"
+    with patch(
+        "homeassistant.components.dynalite.bridge.DynaliteDevices.async_setup",
+        return_value=True,
+    ), patch(
+        "homeassistant.components.dynalite.bridge.DynaliteDevices.available", True
+    ):
+        await hass.config_entries.flow.async_init(
+            dynalite.DOMAIN,
+            context={"source": config_entries.SOURCE_IMPORT},
+            data={dynalite.CONF_HOST: host},
+        )
+        # Do it for a second time
+        result = await hass.config_entries.flow.async_init(
+            dynalite.DOMAIN,
+            context={"source": config_entries.SOURCE_IMPORT},
+            data={dynalite.CONF_HOST: host},
+        )
+    assert result["type"] == "abort"
+    assert result["reason"] == "already_configured"
+
+
+async def test_existing_different_config(hass):
+    """Test when the entry exists with a different config."""
+    host = "1.2.3.4"
+    with patch(
+        "homeassistant.components.dynalite.bridge.DynaliteDevices.async_setup",
+        return_value=True,
+    ), patch(
+        "homeassistant.components.dynalite.bridge.DynaliteDevices.available", True
+    ):
+        await hass.config_entries.flow.async_init(
```

Comment:
Add the first entry as a `MockConfigEntry` instead and use entry.add_to_hass.

https://github.com/home-assistant/home-assistant/blob/f75bda75512e36faaebddcd314149553d99d9f6f/tests/common.py#L656-L694

Reply:
done